src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.cs

```diff
@@ -356,5 +356,16 @@ static bool HasCaseInsensitiveBackReferences(RegexNode node)
                 return false;
             }
         }
+
+        /// <summary>Stores the data necessary to create a Diagnostic.</summary>
+        /// <remarks>
+        /// Diagnostics do not have value equality semantics.  Storing them in an object model
+        /// used in the pipeline can result in unnecessary recompilation.
+        /// </remarks>
+        private sealed record class DiagnosticData(DiagnosticDescriptor descriptor, Location location, object? arg = null)
```

Comment:
Nit: this will need to be rewritten once a diagnostic with > 1 arguments becomes necessary.

Reply:
If that ever happens, yes, this private type will need to be updated.